### ISSUE TITLE
Java Slim's converter for Collections toString() is broken for elements without an explicit Converter

### ISSUE THREAD

**Issue by fhoeben** [unclebob/fitnesse#740]
To convert elements of a generic collection (or array) Slim's converter falls back to using the StringConverter. This works fine for converting from Strings, but it does NOT work for converting TO Strings. As the code will attempt to pass in a non-String object  to a method requiring a String argument (`StringConverter.toString()`).

To demonstrate the issue a slightly updated version of one of the methods of `ConverterRegistryTest`, which will throw an exception on `converter.toString(listToConvert)`

```java
  @Test
  @SuppressWarnings("rawtypes")
  public void getConverterForClass_should_return_a_collection_of_typed_object_without_converter_when_value_is_a_typed_list() throws SecurityException, NoSuchMethodException {
    Class<List> clazz = List.class;

    ParameterizedType typedClass = (ParameterizedType) new ArrayList<Object>() {} .getClass().getGenericSuperclass();

    Converter<List> converter = ConverterRegistry.getConverterForClass(clazz, typedClass);
    Object current = converter.fromString("[1]").get(0);

    assertTrue(converter instanceof GenericCollectionConverter);
    assertTrue(current instanceof String);
    assertEquals("1", current);

    List<Object> listToConvert = new ArrayList<Object>();
    listToConvert.add(this);
    listToConvert.add(new StringBuilderConverter());
    String converted = converter.toString(listToConvert);
    Pattern p = Pattern.compile("\\[(.*?),\\s*(.*?)\\]");
    assertTrue(p.matcher(converted).matches());
  }

```

**Comment by fhoeben** [comment]
Fixed in pull request #739

**Comment by amolenaar** [comment]
PR has been merged
